File: haskell/003/Main.hs
{-
    ProjectEuler 003
    Haskell / Main.hs
    copyright (c) 2015 Susisu

    GHC 7.8.3
    `ghc Main.hs -o bin/main`
-}

module Main where

main :: IO ()
main = print $ calc 600851475143 2

calc :: Integer -> Integer -> Integer
calc n p
    | n == 1         = p
    | n `mod` p == 0 = calc (n `div` p) p
    | otherwise      = calc n (p + 1)

File: lua/003/main.lua
--[[
    ProjectEuler 003
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local n = 600851475143
    local i = 2
    while n > 1 do
        if n % i == 0 then
            n = n / i
        else
            i = i + 1
        end
    end
    print(i)
end

main()
